src/utils/activity-object-helper.ts

```diff
@@ -0,0 +1,114 @@
+import { ActivityObject, Link, Image } from "activitypub-objects";
+import { RdfLangString } from "@/model/rdf-lang-string";
+import { Activity } from "@/model/mitra-activity";
+
+export class ActivityObjectHelper {
+  public static hasProperty(obj: object, property: string): boolean {
+    try {
+      return property in obj;
+    } catch (err) {
+      return false;
+    }
+  }
+
+  public static extractActorName(
+    object: ActivityObject | Link | URL | Array<ActivityObject | URL>
+  ): string | undefined {
+    const lang = navigator.language.substr(0, 2);
+    if (ActivityObjectHelper.hasProperty(object, "name")) {
+      return (object as ActivityObject).name;
+    } else if (ActivityObjectHelper.hasProperty(object, "nameMap")) {
+      return (object as RdfLangString).nameMap[lang];
+    } else if (ActivityObjectHelper.hasProperty(object, "map")) {
```

Comment:
according to the standard an object can not have a property `map` but only `nameMap` if I don't miss anything. Why do you think this if block is needed?